File: ZCMS_Backend/src/main/java/com/devz/zcms/service/ContentService.java
package com.devz.zcms.service;

import com.devz.zcms.entity.Content;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface ContentService {
    Mono<Content> createNewContent(Content content);
    Flux<Content> getAllContentsByContentModelId(String contentModelId);
    Flux<Content> getAllContents();
    Mono<Content> updateContent(Content content);
    Mono<Void> deleteContentByContentId(String id);
}

File: ZCMS_FrontEnd/src/app/service/content.service.ts
import { Injectable } from '@angular/core';
import {SystemConstant} from "../const/SystemConstant";
import {HttpClient, HttpParams} from "@angular/common/http";
import {AuthService} from "./auth.service";
import {Content} from "../model/Content";
import {Observable} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class ContentService {
  private defaultUrl: string = SystemConstant.BASE_PATH + '/contents';

  constructor(private httpClient: HttpClient) { }

  getAllContentsByContentModelId(contentModelId: string): Observable<Content[]>{
    const httpParams = new HttpParams().set("contentModelId", contentModelId);
    return this.httpClient.get<Content[]>(this.defaultUrl, {params: httpParams});
  }

  createNewContent(content: Content): Observable<Content>{
    return this.httpClient.post<Content>(this.defaultUrl, content);
  }

  updateExistingContent(content: Content): Observable<Content>{
    return this.httpClient.put<Content>(this.defaultUrl, content);
  }

  deleteContentById(contentId: string): Observable<any>{
    const url = this.defaultUrl + '/' + contentId;
    return this.httpClient.delete(url);
  }
}
